### PR TITLE
@ #41 | Should allow specifying excluded methods from becoming routes

### LINKED ISSUE #41: Can't Add Public Methods Without Them Becoming Routes
In previous versions (not sure of the range, because I don't know when this behavior was introduced) of Flask Classful, it was possible to have a public method in your view class and not have it automatically registered as a route. Take for example:

```python
from flask import request
from flask_classful import FlaskView

class BaseView(FlaskView):
    def copy_form_data(self):
         return request.form.copy()
```

In this example, `BaseView.copy_form_data` is registered as a route on all views that extend from `BaseView`. IMHO this isn't desired behavior, but whatevs.

So another issue related to this that I discovered (and maybe related to #29) is that I cannot make `BaseView.copy_form_data` into a static method because a `flask_classful.DecoratorCompatibilityError` is raised when it's encountered.

I have found that the only way around this restriction is to make that method private, which isn't super optimal because it makes it tricky to document with Sphinx and that it breaks existing code that depends on it.

### Discussion Points

1. Is it possible to whitelist methods that we don't want to be registered as routes? I would think that we would denote routes with the `route` decorator, but it appears that the ship has sailed on that design decision.
2. Should we automatically exclude static methods from being registered as routes?

### PR DESCRIPTION
**Pull request by hoatle**
connect #41

### CODE REVIEW

File: flask_classful.py [hunk 1]
```diff
@@ -422,7 +423,8 @@ def get_interesting_members(base_class, cls):
             )
             and not member[0].startswith("_")
             and not member[0].startswith("before_")
-            and not member[0].startswith("after_")]
+            and not member[0].startswith("after_")
```
Comment (mattbillenstein) [inline flask_classful.py]: This list comprehension is getting kinda crazy -- recommend re-writing as a regular loop...

**Comment by mattbillenstein** [comment]
I must admit, I kinda prefer prefixing private methods with '_' as this is how you signal a non-public part of the api by convention in Python and you can tell at first glance when looking at a method if it's considered public or not.  This is at least the convention I've been following in my code.

That being said, others may prefer a different style so they may like this better.

File: flask_classful.py (hunk 1 above)
Comment (hoatle) [reply flask_classful.py]: thank you, we'll consider this later.

**Comment by hoatle** [comment]
thank you guys, merged.

**Comment by hjc** [comment]
@mattbillenstein - I totally agree, but I wouldn't think of something you don't want to expose over the API as necessarily always being a private method. Consider some sort of public setup method that the Flask application should call when registering or after registering this View, while this method isn't a private method, it's also not something that should be exposed over an API endpoint. In this case, prefixing it with an `_` isn't appropriate, and neither is registering it as a route.

IIRC, we already don't register methods beginning with an `_` as routes, this just supplements that for more niche use cases.

IMO, this needs a bit of docs, lemme try to write those up real quick.

**Comment by hjc** [comment]
Docs opened in #59.

**Pull request merged**